src/hotspot/cpu/x86/macroAssembler_x86.cpp

```diff
@@ -3480,6 +3480,58 @@ void MacroAssembler::vbroadcastss(XMMRegister dst, AddressLiteral src, int vecto
   }
 }
 
+// Vector float blend
+// vblendvps(XMMRegister dst, XMMRegister nds, XMMRegister src, XMMRegister mask, int vector_len, bool fully_masked = false, XMMRegister scratch = xnoreg)
+void MacroAssembler::vblendvps(XMMRegister dst, XMMRegister src1, XMMRegister src2, XMMRegister mask, int vector_len, bool fully_masked, XMMRegister scratch) {
+  // WARN: Allow dst == (src1|src2), mask == scratch
+  bool scratch_available = scratch != xnoreg && scratch != src1 && scratch != src2 && scratch != dst;
+  bool dst_available = dst != src1 || dst != src2;
```

Comment:
What if it is fully masked and dst==mask? In that case also dst_available should be false.

Reply:
Hmm.. thats a good catch.. but its also bad if `!fully_masked && mask==scratch` (i.e. both cases will destroy the mask before it can be used)

For readability, I think I will just keep a second half of the check: `bool dst_available = dst!=mask && (dst != src1 || dst != src2)` 
(i.e. whomever calls this function, really _should_ know better, just like all the other functions in this file that dont have input validation.. these checks started as asserts while developing to catch some confusing bugs..)